### ISSUE TITLE
Add `changesets` for managing releases?

### ISSUE THREAD

**Issue by TimMikeladze** [huggingface/huggingface.js#6]
https://pnpm.io/using-changesets
https://github.com/changesets/changesets

This is a tool that assists in managing the release cycle for packages in a mono-repo. It will help with generating a changelog, creating github releases, git tags, etc.

**Comment by coyotte508** [comment]
Looks nice! As long as releases can be done directly from github actions like now

**Comment by pngwn** [comment]
Yes. There is a bot: https://github.com/apps/changeset-bot which is nice to help contributors add the correct files and an action: https://github.com/changesets/action which makes every CI driven.

It can also handle pre-releases although that is a touch more involved,

I've used changesets for a few years and found it really helpful for handling releases. I haven't set this up in a while but I'll be doing it for gradio soon, I don't mind having a go here too.

**Comment by o-az** [comment]
I'm happy to help with setting this up too! Been using `changesets` and it's very nice.

**Comment by coyotte508** [comment]
Sure! 

For now we have two github actions, one for each package (could probably be reduced to one, with an additional parameter).

The github action bumps up the version, updates the docs, publishes to NPM and pushes the commit to the repo.

**Comment by pngwn** [comment]
Changesets is designed more multi package repos so should work fine ootb. 

Changesets will create a 'version' PR that gets added to every time you merge to main. Merging that version PR will publish all necessary versions. It's nice in that the action never commits to main, just to a special branch that you then review + merge. I hate CI pushing directly to main, so I like this feature personally.

**Comment by coyotte508** [comment]
Sounds good!
